modules/ibm-z-rhel-kvm-host-recommendations.adoc

```diff
@@ -0,0 +1,181 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/ibm-z-recommended-host-practices.adoc
+
+[id="ibm-z-rhel-kvm-host-recommendations_{context}"]
+= {op-system-base} KVM on IBM Z host recommendations
+
+Optimizing a KVM virtual server environment strongly depends on the workloads of the virtual servers and on the available resources. The same action that enhances performance in one environment can have adverse effects in another. Finding the best balance for a particular setting can be a challenge and often involves experimentation.
+
+The following section introduces some best practices when using OpenShift with {op-system-base} KVM on IBM Z and LinuxONE environments.
+
+== Use multiple queues for your VirtIO network interfaces
+
+With multiple virtual CPUs, you can transfer packages in parallel if you provide multiple queues for incoming and outgoing packets. Use the queues attribute of the driver element to configure multiple queues. Specify an integer of at least 2 that does not exceed the number of virtual CPUs of the virtual server.
+
+The following example specification configures two input and output queues for a network interface:
+
+[source,terminal]
+----
+<interface type="direct">
+    <source network="net01"/>
+    <model type="virtio"/>
+    <driver ... queues="2"/>
+</interface>
+----
+
+Multiple queues are designed to provide enhanced performance for a network interface, but they also use memory and CPU resources. Start with defining two queues for busy interfaces. Next, try two queues for interfaces with less traffic or more than two queues for busy interfaces.
+
+== Use I/O threads for your virtual block devices
+
+To make virtual block devices use I/O threads you must configure one or more I/O threads for the virtual server and each virtual block device to use one of these I/O threads.
+
+The following example specifies `<iothreads>3</iothreads>`  to configure three I/O threads, with consecutive decimal thread IDs 1, 2, and 3. 
+`iothread="2"` specifies the driver element of the disk device to use the I/O thread with ID 2.
+
+
+.Sample I/O thread specification
+[source,terminal]
+...
+<domain>
+ 	<iothreads>3</iothreads><1>
+  	 ...
+    	<devices>
+       ...
+          <disk type="block" device="disk"><2>
+<driver ... iothread="2"/>
+    </disk>
+       ...
+    	</devices>
+   ...
+</domain>
+----
+<1> The number of I/O threads.
+<2> The driver element of the disk device.
+
+Threads can increase the performance of I/O operations for disk devices, but they also use memory and CPU resources. You can configure multiple devices to use the same thread. Which mapping of threads to devices leads to the best results depends on the available resources and the workload.
```

Comment:
```suggestion
Threads can increase the performance of I/O operations for disk devices, but they also use memory and CPU resources. You can configure multiple devices to use the same thread. The best mapping of threads to devices depends on the available resources and the workload.
```
Just a suggestion; the last sentence wasn't super clear to me when I first read it.

Reply:
Yours is a lot better I remember that one caused me some 🧠